### DISCUSSION TITLE
how to use roslyn without using ide to do static code analysis

### DISCUSSION THREAD

**Discussion by alexlee820** [dotnet/roslyn#62817] [Q&A]
I am writing a code analysis platform ,is there any way that we can use Roslyn without using visual studio ,and directly run the code in cmd or powershell to output the analysis result.

**Comment by jasonmalinowski** [comment]
Roslyn absolutely is a library you can call to do various things. Without further description of what yuo'd trying to do, all we can give is a "likely yes" though.
